modules/ingester/trace.go

```diff
@@ -17,20 +30,26 @@ type trace struct {
 	traceID      []byte
 	maxBytes     int
 	currentBytes int
+
+	// List of flatbuffers
+	searchData         [][]byte
+	maxSearchBytes     int
+	currentSearchBytes int
 }
 
-func newTrace(maxBytes int, traceID []byte) *trace {
+func newTrace(traceID []byte, maxBytes int, maxSearchBytes int) *trace {
 	return &trace{
 		traceBytes: &tempopb.TraceBytes{
 			Traces: make([][]byte, 0, 10), // 10 for luck
```

Comment:
should we preallocate `searchData` for luck as well? :)

Reply:
Luck is important :)  But in this case search data is optional, and the ingester may not be receiving it from the distributor.  I think it would be good to pool resources better here in the future. The TraceBytes struct functionally makes sense (alloc up front, return to pool when flushing trace), but is too specific-named.

Reply:
Yeah agree. Let's take these optimisations up in a later PR